tests/core18/snapd-refresh/task.yaml

```diff
@@ -0,0 +1,40 @@
+summary: Test refreshing snapd and running new units
+
+execute: |
+    echo "Testing refresh/revert of snapd"
+    current=$(readlink /snap/snapd/current)
+    SNAPD_SNAP=$(ls /var/lib/snapd/snaps/snapd_"$current".snap)
+    for _ in $(seq 10); do
+
+        echo "Installing a new snapd snap"
+        snap install --dangerous "$SNAPD_SNAP"
+        echo "Still leaves 'snap list' working"
+        if snap list | grep "snapd.*$current "; then
```

Comment:
It would be useful to check if snapd was restarted, eg. check that `systemctl show -p MainPID snapd.service` if different each time.